src/workloads/scale/MemoryStress.yml

```diff
@@ -0,0 +1,103 @@
+SchemaVersion: 2018-07-01
+Owner: Service Architecture
+Description: |
+  Used to test performance of the server under heavy memory pressure. This workload is
+  expected to fail due to host(s) being unreachable as a result of mongod(s) running out
+  of memory. When the server becomes more suited to handle memory pressure, the
+  expectation of failure can be removed.
+
+Keywords:
+- scale
+- memory stress
+- aggregate
+- sort
+- insert
+- fail
+
+GlobalDefaults:
+  dbname: &DBName memorystress
+  MaxPhases: &MaxPhases 20
+
+  LoadThreads: &LoadThreads 128
+  LoadBatchSize: &LoadBatchSize 500
+
+  NumberOfDocuments: &NumDocs 20096
+  # NumberOfDocumentsPerWorker = NumberOfDocuments / LoadThreads.
+  NumberOfDocumentsPerWorker: &DocsPerThread 157
+  DocSize: &DocSize 520000
+  Document: &Doc
+    # Document number/ID. Starts at -*DocsPerThread because ActorId starts at 1, and therefore
+    # in order for the first output of ^Inc to be 0, "start" would need to be offset by
+    # ActorId * *DocsPerThread.
+    a: {^Inc: {start: -157, step: 1, multiplier: *DocsPerThread}}
+    # Random number from [0, NumberOfDocuments].
+    b: {^RandomInt: {min: 0, max: *NumDocs}}
+    # Fill 520 KB.
+    c: {^FastRandomString: {length: *DocSize}}
+
+  # Number of documents to sort by each thread. Tuned so that *SortStep * *DocSize < 100 MB,
+  # which is the maximum memory a sort query is allowed to use before spilling to disk.
+  SortStep: &SortStep 180
+  SortBatchSize: &SortBatchSize 180
+  SortThreads: &SortThreads 250
+  SortRepeat: &SortRepeat 5
+
+Clients:
+  Default:
+    QueryOptions:
+      socketTimeoutMS: -1
+      maxPoolSize: 500
+    
+Actors:
+- Name: LoadDocuments
```

Comment:
Note: I've seen the following [setup phase](https://github.com/mongodb/genny/blob/ba047273ee7b8d5b693ca70dab51d97e4b948bc2/src/workloads/query/GroupSpillToDisk.yml#L23-L37)  added to some workloads. It can make local testing simpler.

Reply:
Done!